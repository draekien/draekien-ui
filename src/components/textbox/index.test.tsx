/** @jsxRumtime classic */
/** @jsx jsx */
import { jsx } from 'theme-ui';
import { render } from '@testing-library/react';
import Textbox from '.';

describe('Textbox component', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <Textbox id="test" label="test" tooltip="test" icon="info" />
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when active and fullwidth', () => {
    const { container } = render(
      <Textbox
        id="test"
        label="test"
        tooltip="test"
        icon="info"
        active
        fullWidth
      />
    );

    expect(container).toMatchSnapshot();
  });
});
