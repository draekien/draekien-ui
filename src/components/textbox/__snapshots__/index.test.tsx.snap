// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox component should match snapshot 1`] = `
<div>
  <div
    class="css-18cm3mp"
  >
    <label
      class="css-159wv0z"
      for="test"
    >
      <div
        class="css-33lc35"
      >
        test
      </div>
    </label>
    <div
      class="css-1mngvxn"
    >
      <input
        class="css-124nka3"
        id="test"
        value=""
      />
      <span
        class="css-1fkmtnz"
      >
        <i
          class="material-icons  css-eyhhi6"
        >
          info
        </i>
      </span>
    </div>
    <label
      class="css-159wv0z"
      for="test"
    >
      <div
        class="css-33lc35"
      >
        test
      </div>
    </label>
  </div>
</div>
`;

exports[`Textbox component should match snapshot when active and fullwidth 1`] = `
<div>
  <div
    class="css-yw2ope"
  >
    <label
      class="css-159wv0z"
      for="test"
    >
      <div
        class="css-33lc35"
      >
        test
      </div>
    </label>
    <div
      class="css-1mngvxn"
    >
      <input
        class="css-rl5gon"
        id="test"
        value=""
      />
      <span
        class="css-1fkmtnz"
      >
        <i
          class="material-icons  css-eyhhi6"
        >
          info
        </i>
      </span>
    </div>
    <label
      class="css-159wv0z"
      for="test"
    >
      <div
        class="css-33lc35"
      >
        test
      </div>
    </label>
  </div>
</div>
`;
